docs/data/charts/pie-demo/OnSeriesItemClick.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { PieChart } from '@mui/x-charts/PieChart';
+import { Typography, Stack } from '@mui/material';
+
+export default function OnSeriesItemClick() {
+  const [clickedItem, setClickedItem] = React.useState<any>();
+  return (
+    <Stack direction={'row'} alignItems={'center'}>
+      <Stack>
+        <Typography variant="h6">
+          {clickedItem
+            ? `Clicked on ${clickedItem.label},`
+            : 'Click on a series item'}
+        </Typography>
+        <Typography variant="h6">
+          {clickedItem ? `value: ${clickedItem.value}` : ''}
+        </Typography>
+      </Stack>
+
+      <PieChart
+        series={[
+          {
+            data: [
+              { id: 0, value: 10, label: 'series A', onClick: setClickedItem },
+              { id: 1, value: 15, label: 'series B', onClick: setClickedItem },
+              { id: 2, value: 20, label: 'series C', onClick: setClickedItem },
+            ],
+          },
+        ]}
+        width={400}
+        height={200}
+      />
```

Comment:
What would you think about this way of doing it? This is how Nivo and Rechart do it, and I think it scales better with other charts.

```suggestion
      <PieChart
        series={[
          {
            data: [
              { id: 0, value: 10, label: 'series A' },
              { id: 1, value: 15, label: 'series B' },
              { id: 2, value: 20, label: 'series C' },
            ],
          },
        ]}
        onClick={setClickedItem}
        width={400}
        height={200}
      />
```

I  think it could return 3 arguments: `PieItemIdentifier` is a small object that identifies the item. The item (what you are already passing) and maybe the `event`

I don't know if `PieItemIdentifier` and the item could be fusion. I such a case, it would be a nice signature `(data, event) => void`

Reply:
I agree that it would be more flexible
I just wonder if we should name it `onItemClik` (or similar) and keep `onClick` for the root node event.

Concerning the signature, the core puts the event on the 1st param, might make sense to follow the same nomenclature here

Reply:
Both recharts and nivo use `onClick`, and in 99% of the use case you want to trigger action on item click. 

If you reach a level of customisation where you need to get click from the surface, you might want to use composition. Then it will be possible to do to the following

```jsx
<ChartContainer onClick={handleSurfaceClick}>
  <BarPlot onClick={handleBarClick} />
  <LinePlot onClick={handleLineClick} />
</ChartContainer>
```

Reply:
Moved onclick handler to root component and added Identifier and event